packages/protocol/test/customHooks.ts

```diff
@@ -0,0 +1,13 @@
+import { retryAsync } from '@celo/utils/lib/async'
+
+// Handles flakey `error: Invalid JSON RPC response: ""` error that seems to be caused by port exhaustion in CI.
+// See https://github.com/ethereum/web3.js/issues/3425 and https://github.com/ethereum/web3.js/issues/926.
+export const beforeEachWithRetries = (
+  title: string,
+  numRetries: number,
+  sleepMs: number,
+  fn: () => any
+) =>
+  beforeEach(title, async () => {
```

Comment:
Optional Nit: This is super minor, but since retryAsync effectively, you should be able to just do 

`beforeEach(title, retryAsync(fn, numRetries, [], sleepMs))`